src/pages/Swap/index.tsx

```diff
@@ -299,8 +299,8 @@ export function Swap({
     [independentField, parsedAmount, showWrap, trade]
   )
 
-  const fiatValueInput = useUSDPrice(parsedAmounts[Field.INPUT])
-  const fiatValueOutput = useUSDPrice(parsedAmounts[Field.OUTPUT])
+  const fiatValueInput = useUSDPrice(parsedAmounts[Field.INPUT], currencies[Field.INPUT] ?? undefined)
```

Comment:
Can we also add this improvement to the swap modal in the Token Details page to maintain parity? see https://app.uniswap.org/#/tokens/ethereum/0x1f9840a85d5af5bf1d1762f925bdaddc4201f984

Reply:
I think they already share code

Reply:
they don't you still have the sequential loading state on details for usd price